command/agent/command.go

```diff
@@ -1081,8 +1081,12 @@ func (c *Command) handleReload(config *Config) (*Config, error) {
 		errs = multierror.Append(errs, fmt.Errorf("Failed unloading checks: %s", err))
 		return nil, errs
 	}
+	if err := c.agent.unloadMetadata(); err != nil {
```

Comment:
Are we doing anything like that with the checks/services during the reload? it seems like they have the same problem at the moment.